pkg/plugins/ca/builtin/manager.go

```diff
@@ -0,0 +1,129 @@
+package builtin
+
+import (
+	"context"
+	"fmt"
+
+	core_model "github.com/Kong/kuma/pkg/core/resources/model"
+
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"github.com/pkg/errors"
+
+	mesh_proto "github.com/Kong/kuma/api/mesh/v1alpha1"
+	system_proto "github.com/Kong/kuma/api/system/v1alpha1"
+	core_ca "github.com/Kong/kuma/pkg/core/ca"
+	ca_issuer "github.com/Kong/kuma/pkg/core/ca/issuer"
+	core_system "github.com/Kong/kuma/pkg/core/resources/apis/system"
+	core_store "github.com/Kong/kuma/pkg/core/resources/store"
+	secret_manager "github.com/Kong/kuma/pkg/core/secrets/manager"
+)
+
+type builtinCaManager struct {
+	secretManager secret_manager.SecretManager
+}
+
+func NewBuiltinCaManager(secretManager secret_manager.SecretManager) core_ca.Manager {
+	return &builtinCaManager{
+		secretManager: secretManager,
+	}
+}
+
+var _ core_ca.Manager = &builtinCaManager{}
+
+func (b *builtinCaManager) Ensure(ctx context.Context, mesh string, backend mesh_proto.CertificateAuthorityBackend) error {
+	_, err := b.getCa(ctx, mesh, backend.Name)
+	if core_store.IsResourceNotFound(err) {
+		if err := b.create(ctx, mesh, backend.Name); err != nil {
+			return errors.Wrapf(err, "failed to create CA for mesh %q and backend %q", mesh, backend.Name)
+		}
+	} else {
+		return err
+	}
+	return nil
+}
+
+func (b *builtinCaManager) ValidateBackend(ctx context.Context, mesh string, backend mesh_proto.CertificateAuthorityBackend) error {
+	return nil // builtin CA has no config
+}
+
+func (b *builtinCaManager) create(ctx context.Context, mesh string, backendName string) error {
+	keyPair, err := newRootCa(mesh)
+	if err != nil {
+		return errors.Wrapf(err, "failed to generate a Root CA cert for Mesh %q", mesh)
+	}
+
+	certSecret := &core_system.SecretResource{
+		Spec: system_proto.Secret{
+			Data: &wrappers.BytesValue{
+				Value: keyPair.CertPEM,
+			},
+		},
+	}
+	if err := b.secretManager.Create(ctx, certSecret, core_store.CreateBy(certSecretResKey(mesh, backendName))); err != nil {
```

Comment:
Probably that doesn't matter but is there any point to store CA certificate as a secret? I think it could be stored as ConfigMap and be available to anyone who is willing to reach it

Reply:
There is no notion of ConfigMap on Universal